server/server.go

```diff
@@ -996,13 +999,17 @@ func (s *Server) startPrometheusServer(listenAddr *net.TCPAddr) *http.Server {
 		ReadHeaderTimeout: 60 * time.Second,
 	}
 
-	go func() {
-		s.logger.Info("Prometheus server started", "addr=", listenAddr.String())
-
-		if err := srv.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+	s.logger.Info("Prometheus server started", "addr=", listenAddr.String())
+	go common.RetryForever(context.Background(), time.Second, func(context.Context) error {
```

Comment:
This shouldn't be retried forever, just return the error and stop execution notifying the operator

Reply:
This would be fine if we could alert on every ERROR log. However, as per my other comment, I think that the dispatcher logs ERROR for user request failures, so it is not practical to alert on ERROR

Reply:
@sergerad raises a good point here, which is: is there a clearly defined 'design' for log levels and what the semantics are?
For example, in my thinking anything logged as an ERROR should be a critical system failure - i.e. a highly irregular situation that should be noticed and diagnosed. It would be inappropriate - as @sergerad suggest - to log a simple API error that's caused by malformed use input; since the system is designed to correctly handle this case, a user error should not be considered a critical failure. In fact, I'm not sure I would even WARN on this.

Reply:
Totally, but that's another topic, we don't have a clear definition of criticalities doesn't mean we can hold the execution forever retrying an error, as it's not an expected behaviour.